overrides/default.nix

```diff
@@ -3755,4 +3722,42 @@ lib.composeManyExtensions [
       };
     }
   )
+  # The following are dependencies of torch >= 2.0.0.
+  # torch doesn't officially support system CUDA, unless you build it yourself.
+  (self: super: lib.genAttrs [
+    "nvidia-cublas-cu11"
+    "nvidia-cublas-cu12"
+    "nvidia-cuda-cupti-cu11"
+    "nvidia-cuda-cupti-cu12"
+    "nvidia-cuda-curand-cu11"
+    "nvidia-cuda-curand-cu12"
+    "nvidia-cuda-nvrtc-cu11"
+    "nvidia-cuda-nvrtc-cu12"
+    "nvidia-cuda-runtime-cu11"
+    "nvidia-cuda-runtime-cu12"
+    "nvidia-cudnn-cu11"
+    "nvidia-cudnn-cu12"
+    "nvidia-cufft-cu11"
+    "nvidia-cufft-cu12"
+    "nvidia-curand-cu11"
+    "nvidia-curand-cu12"
+    "nvidia-cusolver-cu11"
+    "nvidia-cusolver-cu12"
+    "nvidia-cusparse-cu11"
+    "nvidia-cusparse-cu12"
+    "nvidia-nccl-cu11"
+    "nvidia-nccl-cu12"
+    "nvidia-nvjitlink-cu11"
+    "nvidia-nvjitlink-cu12"
+    "nvidia-nvtx-cu11"
+    "nvidia-nvtx-cu12"
+  ] (name: super.${name}.overridePythonAttrs (old: {
+    # 1. Remove __init__.py because all the cuda packages include it
+    # 2. Symlink the cuda libraries to the lib directory so autopatchelf can find them
+    postFixup = ''
+      rm -r $out/${self.python.sitePackages}/nvidia/{__pycache__,__init__.py}
+      ln -sfn $out/${self.python.sitePackages}/nvidia/*/lib/lib*.so* $out/lib
```

Comment:
Maybe this is the wrong approach: there's also a `addAutoPatchelfSearchPath` function that can be used to just add libs for that specific package. But I don't know how to make that work for all of its dependencies without listing them all explicitly.

This approach works at least and it's straightforward.